Content.Server/Explosion/EntitySystems/ExplosionSystem.GridMap.cs

```diff
@@ -0,0 +1,374 @@
+                continue;
+            }
+
+            var xform = EntityManager.GetComponent<TransformComponent>(grid.GridEntityId);
+            var matrix = offsetMatrix * xform.WorldMatrix * targetMatrix;
+            var angle = xform.WorldRotation - targetAngle;
+
+            var (x, y) = angle.RotateVec((tileSize / 4, tileSize / 4));
+
+            foreach (var (tile, dir) in edges)
+            {
+                var center = matrix.Transform(tile);
+
+                if ((dir & NeighborFlag.Cardinal) == 0)
+                {
+                    // this is purely a diagonal edge tile
+                    var newIndex = new Vector2i((int) MathF.Floor(center.X), (int) MathF.Floor(center.Y));
+                    if (!transformedEdges.TryGetValue(newIndex, out var data))
+                    {
+                        data = new();
+                        transformedEdges[newIndex] = data;
+                    }
+
+                    data.BlockingGridEdges.Add(new(default, null, center, angle, tileSize));
+                    continue;
+                }
+
+                // Instead of just mapping the center of the tile, we map for points on that tile. This is basically a
+                // shitty approximation to doing a proper check to get all space-tiles that intersect this grid tile.
+                // Not perfect, but works well enough.
+                transformedTiles.Clear();
+                transformedTiles.Add(new((int) MathF.Floor(center.X + x), (int) MathF.Floor(center.Y + y)));  // center of tile, offset by (0.25, 0.25) in tile coordinates
+                transformedTiles.Add(new((int) MathF.Floor(center.X - y), (int) MathF.Floor(center.Y + x)));  // center offset by (-0.25, 0.25)
+                transformedTiles.Add(new((int) MathF.Floor(center.X - x), (int) MathF.Floor(center.Y - y)));  // offset by (-0.25, -0.25)
+                transformedTiles.Add(new((int) MathF.Floor(center.X + y), (int) MathF.Floor(center.Y - x)));  // offset by (0.25, -0.25)
+
+                foreach (var newIndices in transformedTiles)
+                {
+                    if (!transformedEdges.TryGetValue(newIndices, out var data))
+                    {
+                        data = new();
+                        transformedEdges[newIndices] = data;
+                    }
+                    data.BlockingGridEdges.Add(new(tile, gridToTransform, center, angle, tileSize));
+                }
+            }
+        }
+
+        if (referenceGrid == null)
+            return (transformedEdges, tileSize);
+
+        // finally, we also include the blocking tiles from the reference grid.
+
+        if (_gridEdges.TryGetValue(referenceGrid.Value, out var localEdges))
+        {
+            foreach (var (tile, dir) in localEdges)
+            {
+                // grids cannot overlap, so tile should NEVER be an existing entry.
+                var data = new BlockedSpaceTile();
+                transformedEdges[tile] = data;
```

Comment:
Based on how many comments I get about it this is probably happening someday.

Reply:
And I guess someday I'll have to try find a solution, but today is not that day.

Reply:
On a more serious note: can definitely be done, but I have no idea how performant it'll be and haven't really spent any time trying to figure out details. If its only ever gonna be small grids, or grids on top of special collision less tiles, it probably wouldn't be tooo hard?.

Reply:
Actually it might be easier than I thought? I have a decent idea for how to support planets, and if that's done it might not be that hard to do arbitrary grid stacking for explosions.